subscription_test.go

```diff
@@ -369,3 +370,59 @@ func TestEventStreamEventsReceived(t *testing.T) {
 		assert.Fail(t, "channel was not closed")
 	}
 }
+
+func TestRegisterSSESubscriptionMarksMemberDownAndRetries(t *testing.T) {
+	clientCfg := NewDefaultConfig()
+	// Use non-existent address as first cluster member
+	clientCfg.URL = "http://127.0.0.1:11111"
+	clientCfg.EventsTransport = EventsTransportSSE
+	config := configContainer{client: &clientCfg}
+
+	endpoint := newFakeMarathonEndpoint(t, &config)
+	defer endpoint.Close()
+
+	client := endpoint.Client.(*marathonClient)
+	// Add real server as member to the cluster
+	client.hosts.members = append(client.hosts.members, &member{endpoint: endpoint.Server.httpSrv.URL})
+
+	assert.Equal(t, 2, len(client.hosts.activeMembers()), "There should be 2 active members in the cluster")
```

Comment:
I think we should have a `require.` here, no? Otherwise, it doesn't seem meaningful to check for 1 active member in line 393.